rs/dogecoin/ckdoge/minter/src/candid_api.rs

```diff
@@ -92,3 +93,48 @@ impl From<ic_ckbtc_minter::updates::retrieve_btc::RetrieveBtcWithApprovalError>
         }
     }
 }
+
+#[derive(CandidType, Deserialize)]
+pub struct RetrieveDogeStatusRequest {
+    pub block_index: u64,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, CandidType, Deserialize)]
+pub enum RetrieveDogeStatus {
+    /// The minter has no data for this request.
+    /// The request id is either invalid or too old.
+    Unknown,
+    /// The request is in the batch queue.
+    Pending,
+    /// Waiting for a signature on a transaction satisfy this request.
+    Signing,
+    /// Sending the transaction satisfying this request.
+    Sending { txid: Txid },
+    /// Awaiting for confirmations on the transaction satisfying this request.
+    Submitted { txid: Txid },
+    /// The retrieval amount was too low. Satisfying the request is impossible.
+    AmountTooLow,
+    /// Confirmed a transaction satisfying this request.
+    Confirmed { txid: Txid },
+    /// The retrieve bitcoin request has been reimbursed.
+    Reimbursed(ic_ckbtc_minter::state::ReimbursedDeposit),
+    /// The minter will try to reimburse this transaction.
+    WillReimburse(ic_ckbtc_minter::state::ReimburseDepositTask),
+}
+
+impl From<ic_ckbtc_minter::state::RetrieveBtcStatusV2> for RetrieveDogeStatus {
```

Comment:
nit: They look identical, I wonder why you didn't want to just use a definition `type RetrieveDogeStatus = RetrieveBtcStatusV2`.